### PR TITLE
Feat: add mergeFrom and fromIndex in FaissStore

### PR DESCRIPTION
**Pull request by neebdev**
https://github.com/hwchase17/langchainjs/issues/2136. Added the functionality to merge indexes and initialize from another index in FaissStore.

### CODE REVIEW

**Comment by jacoblee93** [comment]
Thank you!

File: langchain/src/vectorstores/faiss.ts [hunk 1]
```diff
@@ -254,6 +289,21 @@ export class FaissStore extends SaveableVectorStore {
     return instance;
   }
 
+  static async fromIndex(
```
Comment (jacoblee93) [inline langchain/src/vectorstores/faiss.ts:292]: What's the purpose of this method? Wouldn't we just call `.load()`?

**Comment by jacoblee93** [comment]
Looks good! Small question but will merge tonight

File: langchain/src/vectorstores/faiss.ts (hunk 1 above)
Comment (neebdev) [reply langchain/src/vectorstores/faiss.ts:292]: Its not absolutely necessary but I think it provides a cleaner way to clone a previous index without saving on disk.

Comment (neebdev) [reply langchain/src/vectorstores/faiss.ts:292]: One more point, It doesn't make sense to require initializing `fromDocuments` or `fromTexts` before they can merge. So `fromIndex` provides a cleaner way of doing: `const vectorStore = new FaissStore(new FakeEmbeddings(), {}); vectorStore.mergeFrom(...);`. Let me know what you think.

**Comment by jacoblee93** [comment]
Thank you!

**Pull request merged**